packages/common/src/utils/typeUtils.ts

```diff
@@ -8,6 +8,9 @@ export function removeNullable<T>(
 export type NestedNonNullable<T> = {
   [P in keyof T]: NestedNonNullable<NonNullable<T[P]>>
 }
+export type NestedNonNullableNonOptional<T> = {
```

Comment:
lol wild

Reply:
I think I'm going to change `NestedNonNullable` to this. The usages of that look like they would tolerate this as well